dockstore-webservice/src/main/resources/migrations.1.4.0.xml

```diff
@@ -15,4 +15,52 @@
   ~    limitations under the License.
   -->
 <databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd" context="1.4.0">
+    <changeSet author="agduncan" id="74274923472389478923">
+        <addColumn tableName="workflow">
+            <column name="sourcecontrol" type="text">
+            </column>
+        </addColumn>
+
+        <sql dbms="postgresql">
+            update workflow set sourcecontrol='GITHUB' where giturl like '%github%'
+        </sql>
+
+        <sql dbms="postgresql">
+            update workflow set path = CONCAT('github.com/', path) where sourcecontrol='GITHUB'
+        </sql>
+
+        <sql dbms="postgresql">
+            update workflow set sourcecontrol='BITBUCKET' where giturl like '%bitbucket%'
+        </sql>
+
+        <sql dbms="postgresql">
+            update workflow set path = CONCAT('bitbucket.org/', path) where sourcecontrol='BITBUCKET'
+        </sql>
+
+        <sql dbms="postgresql">
+            update workflow set sourcecontrol='GITLAB' where giturl like '%gitlab%'
+        </sql>
+
+        <sql dbms="postgresql">
+            update workflow set path = CONCAT('gitlab.com/', path) where sourcecontrol='GITLAB'
+        </sql>
+
+        <addNotNullConstraint tableName="workflow" columnName="sourcecontrol"/>
+
+        <addUniqueConstraint columnNames="sourcecontrol, organization, repository, workflowname" constraintName="uknlbos7i98icbaql5cyt5bhhy2" tableName="workflow"/>
```

Comment:
This this correct? It seems to overlap with full_workflow_name+partial_workflow_name but I could be wrong

Reply:
The workflow table seems to have both the full_workflow_name+partial_workflow_name and this other constraint. There does seem to be overlap.

Reply:
Ok, so after some investigation together. 
I think we want to drop both `uk_kprrtg54h6rjca5l1navospm8` and `ukkprrtg54h6rjca5l1navospm8` using "if exists"